http3/server.go

```diff
@@ -240,6 +257,15 @@ func (s *Server) handleRequest(sess quic.Session, str quic.Stream, decoder *qpac
 	}
 
 	req.RemoteAddr = sess.RemoteAddr().String()
+
```

Comment:
I think you need to reset the deadline after reading the header (if a deadline was set in the first place), right?